### PR TITLE
Python3 script support

### PR DESCRIPTION
**Pull request by tothegills**
Adds support for running pythons scripts. There is a dependency on the `pycryptodome` python module to provide functionality for decrypting sensitive variables. This requires python 3.4+.

Relates to https://github.com/OctopusDeploy/Issues/issues/5234

### CODE REVIEW

File: source/Calamari.Shared/Integration/FileSystem/TemporaryFile.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public long Size
         }
         public void Dispose()
         {
-            fileSystem.DeleteFile(filePath, FailureOptions.IgnoreFailure);
+            //fileSystem.DeleteFile(filePath, FailureOptions.IgnoreFailure);
```
Comment (matt-richardson) [inline source/Calamari.Shared/Integration/FileSystem/TemporaryFile.cs]: Not sure this was meant to be committed :)

File: source/Calamari.Shared/Integration/Scripting/Python/InstallDependencies.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import sys
+
+def install_missing_dependency(module, package):
+    try:
+        printverbose("Checking for dependency {}".format(package))
+        importlib.import_module(module)
+        printverbose("{} was found".format(package))
+    except:
+        printverbose("Did not find dependency, attempting to install {} for the current user using pip".format(package))
+        subprocess.call([sys.executable, "-m", "pip", "install", package, "--user"])
```
Comment (matt-richardson) [inline source/Calamari.Shared/Integration/Scripting/Python/InstallDependencies.py]: pip isn't always installed by default with python... Not sure if you're handling this case.

File: source/Calamari.Tests/Fixtures/RequiresMinimumPython3VersionAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            this.minor = minor;
+        }
+
+        public void BeforeTest(ITest test)
+        {
+            var python3Version = GetPython3Version();
+            if (python3Version.None())
+            {
+                Assert.Ignore("Requires Python3 to be installed");
```
Comment (matt-richardson) [inline source/Calamari.Tests/Fixtures/RequiresMinimumPython3VersionAttribute.cs]: I prefer `Assert.Inconclusive` for these kind of ones - they show up better for this scenario

**Comment by hnrkndrssn** [comment]
A few notes from our screen sharing demo this morning:
1. Provide a helpful message if we can't for some reason install the crypto dependency (i.e. `pip` not being installed on the target), similar to what we do for IIS [here](https://github.com/OctopusDeploy/Calamari/blob/master/source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1#L38)
2. Variable insertion in the script editor doesn't insert the python specific variable format (i.e. `octopusvariables.get('Octopus.Environment.Name')`)
3. Script console does not yet support python scripts

File: source/Calamari.Shared/Integration/Scripting/Python/InstallDependencies.py (hunk 2 above)
Comment (hnrkndrssn) [reply source/Calamari.Shared/Integration/Scripting/Python/InstallDependencies.py]: Suggested we provide a helpful message with some steps the user can take (similar to what we do for IIS) to ensure that they can run python scripts on their targets (i.e. install `pip` or install the `pycryptodome` package using their package manager of choice on the target)

**Pull request merged**